tsinfer/inference.py

```diff
@@ -667,14 +695,15 @@ def insert_missing_sites(
         # We can't perform parsimony inference if all sites are missing, and there's no
         # point if all non-missing sites are the ancestral state, so skip these cases
         if np.all(np.logical_or(G == tskit.MISSING_DATA, G == anc_value)):
+            metadata = _update_site_metadata(
+                site.metadata, inference_type=constants.INFERENCE_NONE
+            )
+            if schema is None:
+                metadata = _encode_raw_metadata(metadata)
```

Comment:
If `schema is None` here then that is basically the same as the schema being the permissive json schema as the metadata is being json encoded without checking the schema. Would it be better just to default to the permissive schema? Or have I missed something about how this would break old code?

Reply:
To support old code we don't want any schema - we don't want the metadata to be decoded automatically at all. That way analysis code will only break if you explicitly opt-in to using a Schema. We'll want to change this going forward, but this seems like a reasonable path forward.